zwrap/sample.go

```diff
@@ -98,70 +103,62 @@ func (s *sampler) With(fields ...zap.Field) zap.Logger {
 }
 
 func (s *sampler) Check(lvl zap.Level, msg string) *zap.CheckedMessage {
-	if !s.check(lvl, msg) {
-		return nil
+	cm := s.Logger.Check(lvl, msg)
+	switch lvl {
+	case zap.PanicLevel, zap.FatalLevel:
+		return cm
 	}
-	return zap.NewCheckedMessage(s.Logger, lvl, msg)
+	if s.sampled(msg) {
```

Comment:
I think that we still need a call to `cm.OK` here:

1. We don't want calls to `logger.Check(zap.DebugLevel, "foo")` to count against the sampling for `logger.Check(zap.ErrorLevel, "foo")` if debug-level logging is disabled.
2. There's also no need to incur the locking/atomic/map access costs of `sampled` for disabled levels.

Perhaps just this:
```go
if !cm.OK() || s.sampled(msg) {
  return cm
}
```